vowpalwabbit/shared_feature_merger.cc

```diff
@@ -0,0 +1,112 @@
+#include "shared_feature_merger.h"
+#include "cb.h"
+#include "example.h"
+#include "label_dictionary.h"
+#include "learner.h"
+#include "options.h"
+#include "parse_args.h"
+#include "vw.h"
+
+#include <iterator>
+
+namespace VW
+{
+namespace shared_feature_merger
+{
+static const std::vector<std::string> option_strings = {
+    "csoaa_ldf", "wap_ldf", "cb_adf", "explore_eval", "cbify_ldf", "cb_explore_adf", "warm_cb"};
+
+bool use_reduction(config::options_i& options)
+{
+  for (const auto& opt : option_strings)
+  {
+    if (options.was_supplied(opt))
+      return true;
+  }
+  return false;
+}
+
+struct sfm_data
+{
+  LEARNER::multi_learner* base;
```

Comment:
Why do you need a base separate from the base learner?

Reply:
The current design requires us to manipulate the examples before running the finish functions on them, this base is used to call into the original finish if necessary

Reply:
Let's find a way to avoid this ---reporting should not be doubling our conversation.